llama-index-integrations/readers/llama-index-readers-azstorage-blob/llama_index/readers/azstorage_blob/base.py

```diff
@@ -98,17 +98,19 @@ def _download_files_and_extract_metadata(self, temp_dir: str) -> Dict[str, Any]:
             )
 
         for obj in blobs_list:
-            sanitized_file_name = obj.name.replace("/", "-")
+            sanitized_file_name = obj.name.replace("/", "-") if not self.blob else obj
```

Comment:
Won't this mean every blob in the list will have the same file name?

Reply:
no? if the user specify a blob name we don't use obj.name, and if not specify we get the obj.name the list_blobs

```py
      if self.blob:
            blobs_list = [self.blob]
        else:
            blobs_list = container_client.list_blobs(
                self.name_starts_with, self.include
            )
 ```

Reply:
Ah OK, missed that